### PR TITLE
[eas-cli] use new android build credentials provider

### PR DESCRIPTION
**Pull request by quinlanj**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.
- [ ] I've tagged the changelog entry with `[EAS BUILD API]` if it's a part of a breaking change to EAS Build API (only possible when updating `@expo/eas-build-job` package).

# Why

Replace the old Android SetupBuildCredentials with the new one

# Test Plan

- [ ] current tests still pass

### CODE REVIEW

File: packages/eas-cli/src/build/android/build.ts [hunk 1]
```diff
@@ -119,8 +122,17 @@ async function ensureAndroidCredentialsAsync(
     }),
     {
       app: {
+        account: nullthrows(
+          findAccountByName(ctx.commandCtx.user.accounts, ctx.commandCtx.accountName),
+          `You do not have access to account: ${ctx.commandCtx.accountName}`
+        ),
         projectName: ctx.commandCtx.projectName,
-        accountName: ctx.commandCtx.accountName,
+        androidApplicationIdentifier: nullthrows(
```
Comment (dsokal) [inline packages/eas-cli/src/build/android/build.ts:130]: This value from app.json is not correct for generic projects. Actually, we even print a warning if someone is building a generic project and they have `android.package` defined in their app.json (see https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/project/android/applicationId.ts#L35). For generic, we read the application id from native code. 

Take a look at line 70 in this file and use the value returned by the `getOrConfigureApplicationIdAsync` function.

**Review by dsokal — CHANGES_REQUESTED** [review]

File: packages/eas-cli/src/build/android/build.ts (hunk 1 above)
Comment (quinlanj) [reply packages/eas-cli/src/build/android/build.ts]: good catch, thanks 🙏

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/build/android/build.ts [hunk 2]
```diff
@@ -113,14 +115,22 @@ async function ensureAndroidCredentialsAsync(
   if (!shouldProvideCredentials(ctx)) {
     return;
   }
+  const androidApplicationIdentifier = await getOrConfigureApplicationIdAsync(
```
Comment (dsokal) [inline packages/eas-cli/src/build/android/build.ts:118]:
```suggestion
  const androidApplicationIdentifier = getApplicationId(
```
This change is related to https://github.com/expo/eas-cli/pull/431#discussion_r641394021
The application id must be already configured in this context.

**Review by dsokal — APPROVED** [review]

File: packages/eas-cli/src/build/android/build.ts (hunk 2 above)
Comment (dsokal) [reply packages/eas-cli/src/build/android/build.ts:118]: Sorry for making you change it again but in https://github.com/expo/eas-cli/pull/427/files/887877aaf9874f6b77a09f3c8632456d54f98355#r640438388 I meant to use the result of the call on line 70. 
However, when I was reviewing https://github.com/expo/eas-cli/pull/431 I realized that we should change the name of  `getOrConfigureApplicationIdAsync` 😅

**Pull request merged**